.circleci/config.yml

```diff
@@ -41,20 +44,29 @@ jobs:
 
       - restore_cache:
           keys:
-            - -{{ checksum "Makefile.envs" }}-v20201205-
```

Comment:
Why do we want to keep the `core` and `packages` ccache separate? The `packages` ccache should include caches from building various packages? Perhaps we could use the same cache for both but only `save_cache` on `build-packages`? Perhaps keeping them separate is simpler.

Reply:
> The packages ccache should include caches from building various packages?

Yes.

> Perhaps we could use the same cache for both but only save_cache on build-packages? Perhaps keeping them separate is simpler.

I think would I rather prefer to have them separate,
 - first, there are surprisingly many things that could go wrong with ccache (particularly with this little documented emsdk fork we are using), and than you have to debug by, setting,
   ```
   export CCACHE_DEBUG=1
   export CCACHE_DEBUGDIR="/root/ccache-debug/"
   ```
   storing that folder as artifacts, than comparing `*.ccache-input-text` and `*.ccache-log` files for subsequent rebuilds to see why there are cache misses. It's easier to do that if there are no inter-dependencies for cache between jobs.
 - for instance on main, ccache is broken for packages because cache loaded contains only files from core. It's hard to detect that because you see "33MB of ccache loaded" and so you think well maybe that's what's supposed to happen. If caches were seprate you would see "0MB of ccache loaded" and also 0 files in `ccache -s` reports making it more obvious that something is wrong.
 - And more generally, it practical to be able to manually reset only the core cache (which is fast to rebuild) and keep the packages cache unchanged if needed. Which also provides a solution for https://github.com/pyodide/pyodide/pull/1805#discussion_r706508747

Reply:
Oh right, given that `build-core` and `build-packages` don't overlap at all in what they build, there isn't even any benefit to sharing the caches.